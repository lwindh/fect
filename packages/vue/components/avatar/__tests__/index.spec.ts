import { mount } from '@vue/test-utils'
import Avatar from '../index'

const BASE_IMG_URL = 'http://img.com'

describe('Avatar', () => {
  it('should be support square and circle', () => {
    const circle = mount(Avatar, {})
    expect(() => circle.unmount()).not.toThrow()
    const square = mount(Avatar, {
      props: {
        isSquare: true,
      },
    })
    expect(() => square.unmount()).not.toThrow()
  })

  it('should be limited 4 characters', () => {
    const ava = mount(Avatar, {
      props: {
        text: 'XeryYue',
      },
    })
    const text = ava.find('.fect-avatar-text').text()
    expect(text.length).toBeLessThanOrEqual(3)
  })
  it('should be render as Element', () => {
    const normalAva = mount(Avatar, {
      props: {
        src: BASE_IMG_URL,
        size: 'mini',
      },
    })
    expect(normalAva.html()).toMatchSnapshot()
    const textAva = mount(Avatar, {
      props: {
        text: 'test',
      },
    })
    expect(textAva.html()).toMatchSnapshot()
  })

  it('stacked should be work', () => {
    const ava = mount(Avatar, {
      props: {
        stacked: true,
      },
    })
    expect(() => ava.unmount()).not.toThrow()
  })
})
