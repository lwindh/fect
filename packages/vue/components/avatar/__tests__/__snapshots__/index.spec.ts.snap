// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should be render as Element 1`] = `
<div class="fect-avatar mini "><img src="http://img.com" draggable="false">
  <!---->
</div>
`;

exports[`Avatar should be render as Element 2`] = `
<div class="fect-avatar medium ">
  <!----><span class="fect-avatar-text ">test</span>
</div>
`;
